cell_classification/train.py

```diff
@@ -0,0 +1,149 @@
+                n_classes = layer.output_shape[-1]
+                loss[layer.name] = self.prep_loss(n_classes)
+
+        self.model.compile(loss=loss, optimizer=self.optimizer)
+        # prepare folders
+        self.model_dir = os.path.join(
+            os.path.normpath(self.params["path"]), self.params["experiment"]
+        )
+        self.log_dir = os.path.join(self.model_dir, "logs")
+        os.makedirs(self.model_dir, exist_ok=True)
+        os.makedirs(self.log_dir, exist_ok=True)
+        self.model_path = os.path.join(self.model_dir, "{}.h5".format(self.params["experiment"]))
+        self.loss_path = os.path.join(self.model_dir, "{}.npz".format(self.params["experiment"]))
+
+        self.num_gpus = count_gpus()
+        print("Training on", self.num_gpus, "GPUs.")
+
+        self.train_callbacks = get_callbacks(
+            self.model_path,
+            lr_sched=self.lr_sched,
+            tensorboard_log_dir=self.log_dir,
+            save_weights_only=self.num_gpus >= 2,
+            monitor="val_loss",
+            verbose=1,
+        )
+
+    def train(self):
+        """Calls prep functions and starts training loops"""
+        self.prep_data()
+        self.prep_model()
+        self.loss_history = self.model.fit(
+            self.train_dataset,
+            epochs=self.params["num_epochs"],
+            validation_data=self.validation_dataset,
+            callbacks=self.train_callbacks,
+        )
+
+    def prep_loss(self, n_classes):
+        """Prepares the loss function for the model
+        Args:
+            n_classes (int): Number of semantic classes in the dataset
+        Returns:
+            loss_fn (function): Loss function for the model
+        """
+        if n_classes == 1:
+            def loss_fn(y_true, y_pred):
+                return tf.keras.losses.binary_crossentropy(y_true, y_pred, from_logits=False)
+        elif n_classes > 1:
+            def loss_fn(y_true, y_pred):
+                return tf.keras.losses.categorical_crossentropy(y_true, y_pred, from_logits=False)
+        return loss_fn
+
+    @staticmethod
+    def prep_batches(batch):
+        """Preprocess batches for training
+        Args:
+            batch (dict):
+                Dictionary of tensors and strings containing data from a single batch
+        Returns:
+            inputs (tf.Tensor):
+                Batch of images
+            targets (tf.Tensor):
+                Batch of labels
+        """
+        inputs = tf.concat(
+            [batch["mplex_img"], tf.cast(batch["binary_mask"], tf.float32)], axis=-1
+        )
+        targets = batch["marker_activity_mask"]
+        return inputs, targets
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--params", type=str, default="cell_classification/configs/params.toml",
+    )
+    args = parser.parse_args()
+    params = toml.load(args.params)
+    trainer = Trainer(params)
+    trainer.train()
```

Comment:
Did it train successfully? How far did you get in testing things out?

Reply:
Had so much other stuff to do.. I'll deploy it tomorrow to our cluster :|

Reply:
No worries, I wasn't sure if you had already tested it, or if it's possible there's still some small bugs